### ISSUE TITLE
New Install fails due to missing cgroup (v2)

### ISSUE THREAD

**Issue by DonBower** [k3s-io/k3s-ansible#219]
I have 4 brand new Raspberry Pi CM4s, running 64 bit Raspberry Pi OS Lite (no GUI), on the nodes, and with GUI on the master.
all 4 failed to start the service:
`level=fatal msg="failed to find memory cgroup (v2)"`

Your documentation should provide this as a pre-req, or add this code to your `prereq/tasks/main.yml` file:

```
- name: Enable container features
  replace:
    path: /boot/cmdline.txt
    regexp: '^([\w](?!.*\b{{ item }}\b).*)$'
    replace: '\1 {{ item }}'
  with_items:
  - "cgroup_enable=cpuset"
  - "cgroup_memory=1"
  - "cgroup_enable=memory"

- name: Reboot the Machine to have container features in affect
  ansible.builtin.reboot:
    reboot_timeout: 300
```

**Comment by cwayne18** [comment]
It's called out here: https://docs.k3s.io/advanced#raspberry-pi

**Comment by DonBower** [comment]
Modified:

```
- name: Enable container features
  replace:
    path: /boot/cmdline.txt
    regexp: '^([\w](?!.*\b{{ item }}\b).*)$'
    replace: '\1 {{ item }}'
  with_items:
  - "cgroup_memory=1"
  - "cgroup_enable=memory"
  register: boot_cmdline

- name: Reboot the Machine to have container features in affect
  ansible.builtin.reboot:
    reboot_timeout: 300
  when: boot_cmdline.changed
```

**Comment by JoeNyland** [comment]
I've just hit this, too. It's fixed in https://github.com/k3s-io/k3s-ansible/pull/185 but it's been waiting to be merged for over a year 😞  It seems this project is dead.

**Comment by daugherr** [comment]
I experienced this issue yesterday installing on my Turing Pi V2 cluster. I noted that it's in the configuration but it didn't work. I manually added the configuration as the fastest way to resolve the issue.

One other thing... the last instruction in README.md didn't make sense to me. kube.cfg was copied by Ansible to the master node. What didn't make sense to me was the difference between source and target.  As written, wouldn't that copy in the same folder? I didn't do that step. Haven't deployed any containers yet but kubectl get nodes sees all my nodes just fine.
